website/files/models/base.py

```diff
@@ -91,6 +91,19 @@ def restore(self, recursive=True, parent=None):
         TrashedFileNode.remove_one(self)
         return restored
 
+    def get_guid(self, create=False):
+        """Attempt to find a Guid that points to this object.
+        One will be created if requested.
+        :rtype: Guid
+        """
+        try:
+            # Note sometimes multiple GUIDs can exist for
+            # a single object. Just go with the first one
+            return Guid.find(Q('referent', 'eq', self))[0]
+        except IndexError:
+            if not create:
+                return None
+        return Guid.generate(self)
```

Comment:
I think by default, if "create=True" isn't passed to the method that this will return "None" if the guid isn't found.

  Thank you for this comment as I just copied this method from "class StoredFileNode" and didn't think about all the possibilities. I assume that, if a TrashedFileNode doesn't have a guid, under no circumstances would we want to create one.